### ISSUE TITLE
Crash when used with ES6 syntax

### ISSUE THREAD

**Issue by Cortys** [ternjs/tern#675]
The tern server crashes as soon as I move inside the parentheses of a function call or declaration to inspect the inferred parameter types.

**Stack trace:**
```
tern/lib/infer.js:1531 self.propagate(new HasMethodCall(pName, args, node.arguments, out));
                                     ^ TypeError: self.propagate is not a function
  at tern/lib/infer.js:1531:14
  at tern/lib/infer.js:1359:7
  at infer (tern/lib/infer.js:1607:22)
  at Object.exports.inferWrapper.walk.make.Expression (tern/lib/infer.js:1616:7)
  at c (tern/node_modules/acorn/dist/walk.js:68:35)
  at Object.base.ExpressionStatement.base.ParenthesizedExpression (tern/node_modules/acorn/dist/walk.js:186:10) 
  at c (tern/node_modules/acorn/dist/walk.js:68:35)
  at Object.exports.inferWrapper.walk.make.Statement (tern/lib/infer.js:1628:7)
  at c (tern/node_modules/acorn/dist/walk.js:68:35)
  at Object.base.Program.base.BlockStatement (tern/node_modules/acorn/dist/walk.js:180:5)
```

**`.tern-project` file:**
```json
{
	"ecmaVersion": 6,
	"libs": [],
	"loadEagerly": [
		"src/**/*.js"
	],
	"plugins": {
		"doc_comment": {
			"fullDocs": true
		},
		"node": {},
		"complete_strings": {}
	}
}
```

This only happens in larger projects and apparently only when using ES6 syntax. I was not able to quickly put together a small staged example where this happens but I can reliably reproduce this issue when working with [this](/runnr/runnr) repository. The issue is reproducible at pretty much every js file in there.

I'm using tern in Atom 1.1.0 on OS X 10.11.1 with the latest version of the atom-ternjs extension (0.8.4 as of writing) and since tern is included in there via git instead of npm I made sure that the currently latest commit https://github.com/ternjs/tern/commit/5e9b6728219ab443a7dd85222f6e06458bb1b5f1 is installed. Idk if it matters but the globally installed node version is 5.0.0 with npm 3.3.6.

I already reported this issue at the atom-ternjs repo https://github.com/tststs/atom-ternjs/issues/155 but it looks like this is an issue with tern itself.

If you need any further information please let me know.

**Comment by marijnh** [comment]
I cloned the repository, npm installed it, and ran some Tern actions on src/app.js, but no crash happened. Are you on the latest version? Anything else that might be relevant?

**Comment by Cortys** [comment]
After restarting Atom yesterday I could not reproduce the problem anymore either, which is odd since I restarted my system, the editor and the tern server multiple times without success since I first encountered the problem. I have no idea what might have changed since then.

So I guess this issue is somehow "resolved" now. I'll post an update if I encounter it again.

Thanks anyways for looking into this! I really appreciate your work.
